pkgs/populate/default.nix

```diff
@@ -145,8 +145,9 @@ let
   populate = target: name: source: let
     source' = source.${source.type};
     target' = target // { path = "${target.path}/${name}"; };
-  in writeDash "populate.${target'.host}.${name}" ''
+  in writeDash "populate.${target'.host}.${replaceChars ["/"] ["_"] name}" ''
     set -efu
+    ${shell' target /* sh */ "mkdir -p $(dirname ${target'.path})"}
```

Comment:
If we really want to support subdirectories as targets, we will have to consider multiple the semantics of following:

```
{
  "foo".file = ...;
  "foo/a".file = ...;
  "foo/b".file = ...;
}
```

Silently failing/clobbering the foo target doesn't sound nice :)

Reply:
I don't see why this definition would lead to a problem, this line only makes sure that the target directory exists and should not conflict with the target directory being a populate target itself.
The only risk I see is foo/a and foo/b being deleted when foo is synced.

Reply:
Not having `foo/a` on the target even though it got specified can be seen as silent failing. Also there might be more unexpected behavior, e.g. both `<foo/a>` and `<foo_a>` can be used to reference `"foo/a"`. But the former is only possible if `"foo".file` is specified, which could delete `"foo/a"` (i'm not sure at the moment which one will be synced first, `"foo"` or `"foo/a"`). Another potential issue could be if `"foo".file` itself contains some `a`. Then `<foo/a>` and `<foo_a>` would potentially point to the wrong one.

Reply:
(We can ignore `"foo/b".file = ...;` from my initial example, I don't see a conflict there either, but I do between `"foo"` and `"foo/a"`.)

Reply:
So the main issue would be fixed by sorting the populate scripts so that foo is always run before foo/a.

Reply:
Yes, looks like that's the main issue. Another issue would be that `"foo/a"` would have to be recreated every run.